File: dataset/Annotation.m
classdef Annotation < handle
    properties
        coordinateList;
        type;
        charId;
        charNameText;
        charState;
        charStateText;
        lineNumber;
        mediaId;
        pathname;
    end
    
    methods
        function obj = Annotation(info, lineNumber,mediaId, pathname)
            parts = strsplit(info,':');
            fprintf('annotation info: %s\n',info);
            obj.coordinateList = char(parts(1));
            obj.type = obj.deriveType(obj.coordinateList);
            obj.charId = char(parts(2));
            obj.charNameText = char(parts(3));
            obj.charState = char(parts(4));
            obj.charStateText = char(parts(5));
            obj.lineNumber = lineNumber;
            obj.mediaId = mediaId;
            obj.pathname = pathname;
        end
        
        function type = deriveType(obj, coordinateList)
            parts = strsplit(coordinateList,';');
            if length(parts) == 1
                type = 'point';
            elseif length(parts) == 2
                type = 'box';    
            else
                type = 'polygon';
            end
        end
       
    end
end

File: java/src/main/java/edu/oregonstate/eecs/iis/avatolcv/Annotation.java
package edu.oregonstate.eecs.iis.avatolcv;

public class Annotation {
	public static final String ANNOTATION_DELIM = "|";
    private String coordinateList;
    private String type;
    private String charId;
    private String charNameText;
    private String charState;
    private String charStateText;
    private int lineNumber;
    private String mediaId;
    private String pathname;
    public Annotation(String info, int lineNumber, String mediaId, String pathname){
    	String[] parts = info.split(":");
    	System.out.println("annotation info: " + info);
        
        this.coordinateList = parts[0];
        this.type = deriveType(this.coordinateList);
        this.charId = parts[1];
        this.charNameText = parts[2];
        this.charState = parts[3];
        this.charStateText = parts[4];
        this.lineNumber = lineNumber;
        this.mediaId = mediaId;
        this.pathname = pathname;
    }
    public String deriveType(String coordinateList){
    	String[] parts = coordinateList.split(";");
        if (parts.length == 1){
        	return "point";
        }  
        else if (parts.length == 2){
            return "box";  
        }
        else{
            return "polygon";
        }
    }
    public String getType(){
    	return this.type;
    }
    public String getCharId(){
    	return this.charId;
    }
    public String getCharNameText(){
    	return this.charNameText;
    }
    public String getCharState(){
    	return this.charState;
    }
    public String getCharStateText(){
    	return this.charStateText;
    }
    public int getLineNumber(){
    	return this.lineNumber;
    }
    public String getMediaId(){
    	return this.mediaId;
    }
    public String getPathname(){
    	return this.pathname;
    }
    public String getTrainingDataLine(String mediaFilename, String taxonId){
    	return "training_data" + ANNOTATION_DELIM + "media/" + mediaFilename + ANNOTATION_DELIM + 
                    charState + ANNOTATION_DELIM + charStateText + ANNOTATION_DELIM + pathname + ANNOTATION_DELIM + taxonId + ANNOTATION_DELIM + lineNumber;          
    }
}
